File: backend/src/models/Diocese.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Diocese extends Model
{
    protected $table = 'Diocese';
    protected $primaryKey = 'DioceseID';
    public $timestamps = false;

    // Updated fillable: remove old 'DioceseState' and add 'StateID'
    protected $fillable = [
        'DioceseName',
        'DioceseStreetNo',
        'DioceseStreetName',
        'DioceseSuburb',
        'StateID',           // numeric foreign key to State
        'DiocesePostcode',
        'DiocesePhone',
        'DioceseEmail',
        'DioceseWebsite',
    ];

    // Relationship: One Diocese belongsTo One State
    public function state()
    {
        return $this->belongsTo(State::class, 'StateID', 'StateID');
    }

    // Relationship: One Diocese has Many Parishes
    public function parishes()
    {
        return $this->hasMany(Parish::class, 'DioceseID', 'DioceseID');
    }

    // Relationship: One Diocese has Many Adorations
    public function adorations()
    {
        return $this->hasMany(\App\Models\Adoration::class, 'DioceseID', 'DioceseID');
    }

    // Relationship: One Diocese has Many Crusades
    public function crusades()
    {
        return $this->hasMany(\App\Models\Crusade::class, 'DioceseID', 'DioceseID');
    }
}

File: backend/src/main/java/com/nearchrist/backend/entity/Diocese.java
package com.nearchrist.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name = "dioceses")
public class Diocese {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long dioceseId;
    @Column(nullable = false)
    private String dioceseName;
    private String dioceseStreetNo;
    private String dioceseStreetName;
    private String dioceseSuburb;
    private String diocesePostcode;
    private String diocesePhone;
    private String dioceseEmail;
    private String dioceseWebsite;
    @OneToMany(mappedBy = "diocese", cascade = CascadeType.ALL)
    @JsonManagedReference("diocese-parishes")
    @JsonIgnoreProperties("diocese")
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private List<Parish> parishes;
    @OneToMany(mappedBy = "diocese", cascade = CascadeType.ALL)
    @JsonManagedReference("diocese-adorations")
    private List<Adoration> adorations;
    @OneToMany(mappedBy = "diocese", cascade = CascadeType.ALL)
    @JsonManagedReference("diocese-crusades")
    private List<Crusade> crusades;
}
